### DISCUSSION TITLE
Question about splitting batches in 'dp' mode

### DISCUSSION THREAD

**Discussion by xiaopi-ouo** [Lightning-AI/pytorch-lightning#5802] [DDP / multi-GPU / multi-node]
Hello, I faced the same problem as mentioned as #1508.
```
    def collate_fn(self, samples):
        output = {}

        output['labels'] = [sample['labels'] for sample in samples]
```
Here the the `samples` is a list of tensor with different size.
It's unavoidable in my case. (each sample actually has different number of parts to be predicted)
So, I can't solved it by just converting them into a tensor `torch.tensor([sample['labels'] for sample in samples])`
Is there other solution?

Thanks.

**Comment by awaelchli** [comment]
I don't think this is the same issue as you linked. 
You want to collate samples of different length into a tensor batch, right?
For this you would need some sort of padding. Perhaps you can hack it easily with
https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pad_sequence.html#torch.nn.utils.rnn.pad_sequence
not sure, have never used it. Good luck :)

**Comment by xiaopi-ouo** [comment]
Thank you:) I finally solved it by padding manually
